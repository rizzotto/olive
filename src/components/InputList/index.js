import React, { useState, useRef } from 'react';
import { Checkbox, IconButton } from '@material-ui/core';
import useStyles from './styles';
import { FiTrash } from 'react-icons/fi';
import Input from '../Input';
import theme from '../../constants/theme';
import PropTypes from 'prop-types';
import Button from '../Button';
function InputList({ checkbox, deleteButton, items, onChange }) {
  const classes = useStyles();
  const [itemsState, setItemsState] = useState(items);
  let timeoutRef = useRef(null);

  function handleNewIngredientClick() {
    setItemsState([...itemsState, '']);
  }

  function handleDeleteButtonClick(id) {
    const newItems = itemsState.filter((item, index) => index !== id);
    setItemsState(newItems);
    onChange(newItems);
  }

  function handleItemChange(event, id) {
    clearTimeout(timeoutRef);
    const newItems = itemsState;
    newItems[id] = event.target.value;
    setItemsState(newItems);
    timeoutRef = setTimeout(function () {
      onChange(newItems);
    }, 1100);
  }

  function clearList() {
    setItemsState([]);
    onChange([]);
  }

  return (
    <div className={classes.list}>
      {itemsState.length !== 0 ? (
        <div className={classes.container}>
          {itemsState.map((item, index) => {
            return (
              <div key={`${item}-${index}`} className={classes.row}>
                {checkbox && (
                  <Checkbox style={{ paddingLeft: 0, paddingRight: 5 }} />
                )}
                <Input
                  placeholder="Novo Ingrediente"
                  hiddenLabel
                  defaultValue={item}
                  onChange={e => handleItemChange(e, index)}
                  autoFocus
                />
                {deleteButton && (
                  <IconButton
                    aria-label="delete"
                    onClick={() => handleDeleteButtonClick(index)}
                    style={{ paddingRight: 0, paddingLeft: 5 }}
                  >
                    <FiTrash color={theme.palette.secondary.main} />
                  </IconButton>
                )}
                <br />
              </div>
            );
          })}
        </div>
      ) : (
        <div className={classes.warningContainer}>
          <span className={classes.warning}>Adicione Novos Ingredientes</span>
        </div>
      )}

      <Button
        variant="outlined"
        color="primary"
        style={{ width: 220, marginBottom: 5 }}
        onClick={() => handleNewIngredientClick()}
        text="Novo Ingrediente"
      />
      {itemsState.length > 0 && (
        <Button
          variant="outlined"
          color="primary"
          style={{ width: 220 }}
          onClick={() => clearList()}
          text="Limpar Lista"
        />
      )}
    </div>
  );
}

InputList.propTypes = {
  checkbox: PropTypes.bool,
  shopList: PropTypes.bool,
  deleteButton: PropTypes.bool,
  itemsProp: PropTypes.arrayOf(PropTypes.string),
  onChange: PropTypes.func,
};

InputList.defaultProps = {
  checkbox: false,
  shopList: false,
  deleteButton: true,
  items: [],
  onChange: () => {},
};

export default InputList;
