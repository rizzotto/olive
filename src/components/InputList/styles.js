import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  list: {
    textAlign: 'center',
  },
  row: {
    display: 'flex',
    width: '100%',
    marginBottom: 15,
  },
  container: {
    overflowY: 'auto',
    height: '100%',
    marginBottom: '10px',
  },
  warningContainer: {
    marginTop: '30px',
    marginBottom: '30px',
  },
  warning: {
    fontWeight: 'bold',
    color: theme.palette.grey.darker,
  },

  '& .MuiInput-input': {
    backgroundColor: theme.typography.primary,
  },
}));

export default useStyles;
