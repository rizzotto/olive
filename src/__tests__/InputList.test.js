import React from 'react';
import renderer from 'react-test-renderer';
import InputList from '../components/InputList';

test('<InputList /> should render list when items prop is not empty', () => {
  const tree = renderer
    .create(<InputList items={['Foo', 'Bar', 'Zedd']} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('<InputList /> should render no item when items prop is empty', () => {
  const tree = renderer.create(<InputList />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<InputList /> should render checkboxes when checkbox prop is defined', () => {
  const tree = renderer
    .create(<InputList checkbox items={['Foo', 'Bar', 'Zedd']} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
