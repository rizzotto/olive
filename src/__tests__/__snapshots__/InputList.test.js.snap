// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputList /> should render checkboxes when checkbox prop is defined 1`] = `
<div
  className="makeStyles-list-1"
>
  <div
    className="makeStyles-container-3"
  >
    <div
      className="makeStyles-row-2"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-input-4 MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
            defaultValue="Foo"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Novo Ingrediente"
            required={false}
            rows={1}
            type="text"
          />
        </div>
      </div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            color="#52944F"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": "#52944F",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
          </svg>
        </span>
      </button>
      <br />
    </div>
    <div
      className="makeStyles-row-2"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-input-4 MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
            defaultValue="Bar"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Novo Ingrediente"
            required={false}
            rows={1}
            type="text"
          />
        </div>
      </div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            color="#52944F"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": "#52944F",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
          </svg>
        </span>
      </button>
      <br />
    </div>
    <div
      className="makeStyles-row-2"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-input-4 MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
            defaultValue="Zedd"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Novo Ingrediente"
            required={false}
            rows={1}
            type="text"
          />
        </div>
      </div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            color="#52944F"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": "#52944F",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
          </svg>
        </span>
      </button>
      <br />
    </div>
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Novo Ingrediente
      </span>
    </button>
  </div>
</div>
`;

exports[`<InputList /> should render list when items prop is not empty 1`] = `
<div
  className="makeStyles-list-1"
>
  <div
    className="makeStyles-container-3"
  >
    <div
      className="makeStyles-row-2"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-input-4 MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
            defaultValue="Foo"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Novo Ingrediente"
            required={false}
            rows={1}
            type="text"
          />
        </div>
      </div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            color="#52944F"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": "#52944F",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
          </svg>
        </span>
      </button>
      <br />
    </div>
    <div
      className="makeStyles-row-2"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-input-4 MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
            defaultValue="Bar"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Novo Ingrediente"
            required={false}
            rows={1}
            type="text"
          />
        </div>
      </div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            color="#52944F"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": "#52944F",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
          </svg>
        </span>
      </button>
      <br />
    </div>
    <div
      className="makeStyles-row-2"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-input-4 MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
            defaultValue="Zedd"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Novo Ingrediente"
            required={false}
            rows={1}
            type="text"
          />
        </div>
      </div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            color="#52944F"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": "#52944F",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
          </svg>
        </span>
      </button>
      <br />
    </div>
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Novo Ingrediente
      </span>
    </button>
  </div>
</div>
`;

exports[`<InputList /> should render no item when items prop is empty 1`] = `
<div
  className="makeStyles-list-1"
>
  <h1>
    Adicione Novos Ingredientes
  </h1>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Novo Ingrediente
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
